sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/broker/SempBasicAuthClientExecutor.java

```diff
@@ -81,6 +87,15 @@ class SempBasicAuthClientExecutor implements Serializable {
     COOKIE_MANAGER_MAP.putIfAbsent(this.cookieManagerKey, new CookieManager());
```

Comment:
Not introduced by this PR, but this map may grow indefinitely. Also, this line unconditionally constructs a `CookieManager`. Could you change this to `COOKIE_MANAGER_MAP.computeIfAbsent(this.cookieManagerKey, CookieManager::new)` so it only constructs a new `CookieManager` when the key is absent?

Reply:
I'll change to `COOKIE_MANAGER_MAP.computeIfAbsent(this.cookieManagerKey, CookieManager::new)`, thanks.
 
Regarding the unconditional growth - in theory yes, but in practice there won't be indefinite number of different combinations of `baseUrls` and `usernames` (`CookieManagerKey`s), so it won't grow beyond a few elements.